### PR TITLE
refactor: legal checkbox test

### PR DESCRIPTION
**Pull request by magiziz**
# Description

Remove current legal checkbox e2e test and replace it with unit test

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/scaffold-ui/package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "url": "https://github.com/reown-com/appkit/issues"
   },
   "devDependencies": {
+    "@open-wc/testing": "4.0.0",
```
Comment (tomiir) [inline packages/scaffold-ui/package.json:50]: aren't you missing `vitest`?

File: packages/scaffold-ui/vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// vitest.config.ts
```
Comment (tomiir) [inline packages/scaffold-ui/vitest.config.ts:1]: remove comment?

File: packages/scaffold-ui/package.json (hunk 1 above)
Comment (magiziz) [reply packages/scaffold-ui/package.json:50]: i think we already have `vitest`. I can see it in `package.json`

**Review by tomiir — APPROVED** [review]

**Comment by lukaisailovic** [comment]
@magiziz Can we also do a click event on the checkbox and asset state in the `OptionsController` based on that?

**Pull request merged**